src/components/Card.js

```diff
@@ -1,28 +1,73 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleLikeDelete
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleLikeDelete = handleLikeDelete;
   }
 
   _setEventListeners() {
     this._cardImageElement.addEventListener("click", () =>
       this._handleImageClick({ name: this._name, link: this._link })
     );
-    this._likeButton.addEventListener("click", () => this._handleLikeButton());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteButton()
-    );
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeButton();
+    });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
   }
 
-  _handleDeleteButton() {
+  remove() {
     this._cardElement.remove();
-    this._cardElement = null;
+    // this._cardElement = null;
   }
 
   _handleLikeButton() {
-    this._likeButton.classList.toggle("card__like-button_active");
+    if (!this._isLiked) {
+      this._handleLikeClick(this._id)
```

Comment:
Needs correction: the Card class should not know anything about the promises.

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```

Reply:
Is there a particular reason for this? I don't remember learning about that or seeing that in the code checklist.

Reply:
The class should not know anything about how the handler is implemented. It can return a promise or it can return anything else.